src/libsyntax/parse/literal.rs

```diff
@@ -16,11 +16,64 @@ use syntax_pos::Span;
 
 use std::ascii;
 
-macro_rules! err {
-    ($opt_diag:expr, |$span:ident, $diag:ident| $($body:tt)*) => {
-        match $opt_diag {
-            Some(($span, $diag)) => { $($body)* }
-            None => return None,
+crate enum LitError {
+    NotLiteral,
+    LexerError,
```

Comment:
We could use `LexerError(ErrorReported)` for clarity here, though I am not sure if it will make things clearer or more confusing :)

Reply:
I did that initially (or rather `LexerError(EscapeError)`), but it ended up unused, so I removed it.
I'll better add a comment.